x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/optimizer/Optimizer.java

```diff
@@ -1197,20 +1231,44 @@ protected LogicalPlan rule(LogicalPlan plan) {
                     }
                     // not everything is foldable, bail out early
                     else {
-                        return values;
+                        return null;
                     }
                 } else if (n.foldable()) {
                     values.add(n.fold());
                 } else {
                     // not everything is foldable, bail-out early
-                    return values;
+                    return null;
                 }
             }
             return values;
         }
 
     }
 
+    static class SkipQueryForOnlyConstantAggregations extends OptimizerRule<Aggregate> {
+
+        @Override
+        protected LogicalPlan rule(Aggregate plan) {
+            if (plan.groupings().isEmpty()
+                && plan.child() instanceof EsRelation
+                && plan.aggregates().stream().allMatch(this::foldable)) {
+
+                return plan.replaceChildrenSameSize(singletonList(new LocalRelation(plan.source(), new SingletonExecutable())));
+            }
+
+            return plan;
+        }
+
+        private boolean foldable(Expression e) {
+            if(e instanceof Alias){
+                return foldable(((Alias) e).child());
+            } else {
+                return e.foldable();
+            }
```

Comment:
This code is familiar - I wonder if it's somewhere else (could be moved into `Expressions`).
Small nit, if you make it a one liner you can put it directly in the if statement.
```suggestion
return (e instanceof Alias ? foldable((Alias) e).child()) : e.foldable());
```

Reply:
Yes, there are a few places where we do the alias resolution manually but the methods that are called in the end are always a bit different. So if we would put it into `Expression` it would need to be fairly generic. Something like `T onResolved(Function<Expression, T> fn)` that could be used like `e.onResolved(Expression::foldable)`.

Alternatively, it might be possible to resolve the aliases in `fold` and `foldable`.